src/integer_mod_q/poly_over_zq/from.rs

```diff
@@ -73,7 +116,44 @@ impl<Mod: Into<Modulus>> From<(&PolyOverZ, Mod)> for PolyOverZq {
     /// # Panics ...
     /// - if `modulus` is smaller than `2`.
     fn from((poly, modulus): (&PolyOverZ, Mod)) -> Self {
-        let mut res = PolyOverZq::from(&modulus.into());
+        let mut res = PolyOverZq::from(modulus);
+        unsafe {
+            fmpz_mod_poly_set_fmpz_poly(
+                &mut res.poly,
+                &poly.poly,
+                res.modulus.get_fmpz_mod_ctx_struct(),
+            );
+        }
+        res
+    }
+}
+
+impl<Mod: Into<Modulus>> From<(PolyOverZ, Mod)> for PolyOverZq {
+    /// Creates a [`PolyOverZq`] from a [`PolyOverZ`] and [`Modulus`].
+    ///
+    /// Parameters:
+    /// - `poly`: the coefficients of the polynomial.
+    /// - `modulus`: the modulus by which each entry is reduced.
+    ///
+    /// # Examples:
+    /// ```
+    /// use qfall_math::integer_mod_q::{PolyOverZq, Modulus};
+    /// use qfall_math::integer::PolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let poly = PolyOverZ::from_str("4  0 1 102 3").unwrap();
+    /// let modulus = Modulus::from(100);
+    ///
+    /// let mod_poly = PolyOverZq::from((poly, &modulus));
+    ///
+    /// # let cmp_poly = PolyOverZq::from_str("4  0 1 2 3 mod 100").unwrap();
+    /// # assert_eq!(cmp_poly, mod_poly);
+    /// ```
```

Comment:
```suggestion
    /// ```
    /// use qfall_math::integer_mod_q::PolyOverZq;
    /// use qfall_math::integer::PolyOverZ;
    /// use std::str::FromStr;
    ///
    /// let poly = PolyOverZ::from_str("4  0 1 102 3").unwrap();
    ///
    /// let mod_poly = PolyOverZq::from((poly, 100));
    ///
    /// # let cmp_poly = PolyOverZq::from_str("4  0 1 2 3 mod 100").unwrap();
    /// # assert_eq!(cmp_poly, mod_poly);
    /// ```
```
Shows the advantage of your function more clearly!